not return error when unmarshalling invalid  base64-encoded string to byte slice

not return an error when unmarshalling invalid  base64-encoded string to a byte slice

code:
```
func TestUnmarshalInvalidBase64EncodedString(t *testing.T) {
    var obj []byte
    data := `"123456"`   // also `"12345"`  or `"1234;6"` 
    err := Unmarshal([]byte(data), &obj)
    require.Error(t, err)
}
```

output:
```
        	Error:      	An error is expected but got nil.
        	Test:       	TestUnmarshalInvalidBase64EncodedString
--- FAIL: TestUnmarshalInvalidBase64EncodedString (0.00s)
```

--- a/go.mod
+++ b/go.mod
@@ -3,7 +3,7 @@ module github.com/bytedance/sonic
 go 1.15
 
 require (
-	github.com/chenzhuoyu/base64x v0.0.0-20210823082418-56861234f7ea
+	github.com/chenzhuoyu/base64x v0.0.0-20211019084208-fb5309c8db06
 	github.com/davecgh/go-spew v1.1.1
 	github.com/goccy/go-json v0.7.2
 	github.com/json-iterator/go v1.1.10

--- a/issue_test/issue112_test.go
+++ b/issue_test/issue112_test.go
@@ -1,0 +1,37 @@
+/*
+ * Copyright 2021 ByteDance Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ package issue_test
+
+ import (
+    `testing`
+
+    . `github.com/bytedance/sonic`
+    `github.com/stretchr/testify/require`
+ )
+ 
+func TestUnmarshalInvalidBase64EncodedString(t *testing.T) {
+    var obj []byte
+    data := `"123456"`
+    err := Unmarshal([]byte(data), &obj)
+    require.Error(t, err)
+
+    data = `"1234;"`
+    err = Unmarshal([]byte(data), &obj)
+    require.Error(t, err)
+}
+ 
+ 

